### PR TITLE
[Jobs] Add integration test for JobInfo GCS endpoint

### PR DESCRIPTION
**Pull request by architkulkarni**
Signed-off-by: Archit Kulkarni <architkulkarni@users.noreply.github.com>

## Why are these changes needed?
Adds an integration test for the GCS endpoint `list_all_jobs` that checks that the JobInfo is propagated correctly from the Ray Job API.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/modules/job/tests/test_job_manager.py [hunk 1]
```diff
@@ -115,6 +116,53 @@ async def test_submit_no_ray_address(call_ray_start, tmp_path):  # noqa: F811
...
+    )
+
+    # Wait for the job to be finished.
+    await async_wait_for_condition_async_predicate(
+        check_job_succeeded, job_manager=job_manager, job_id=submission_id
+    )
+
+    found = False
+    for job_table_entry in (await gcs_aio_client.get_all_job_info()).job_info_list:
+        if job_table_entry.config.metadata.get(JOB_ID_METADATA_KEY) == submission_id:
```
Comment (edoakes) [inline dashboard/modules/job/tests/test_job_manager.py:145]: not really sure which is more clear, but you can also use the `for: else:` syntax to achieve this in python:

`for ...:
    if condition: break
else:
    raise RuntimeError("thing not found")
```

(the `else` block only executes if the loop exits normally)

**Review by edoakes — APPROVED** [review]

**Comment by architkulkarni** [comment]
Debug wheel build failure 



+ echo 'Wheel ray-3.0.0.dev0+dbg-cp310-cp310-manylinux2014_x86_64.whl has the correct commit: 015467d9db9360df067028d9c33b5fc254d91b63'
 

<br class="Apple-interchange-newline">

Known failure on master, unrelated.

Doc build failure unrelated (this PR just updates a test)

File: dashboard/modules/job/tests/test_job_manager.py (hunk 1 above)
Comment (architkulkarni) [reply dashboard/modules/job/tests/test_job_manager.py:145]: Interesting! For-else looks weird to my eye for now, but I may start using it in the future

**Comment by architkulkarni** [comment]
(cc @gvspraveen )

File: dashboard/modules/job/tests/test_job_manager.py (hunk 1 above)
Comment (edoakes) [reply dashboard/modules/job/tests/test_job_manager.py:145]: I used to think so too and at some point I started liking it...

**Pull request merged**